helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -376,50 +658,131 @@ public Object removeRoutingTableChangeListener(
    * @return
    */
   public Collection<LiveInstance> getLiveInstances() {
-    return _routingTableRef.get().getLiveInstances();
+    // Since line instances will be the same across all _routingTableRefMap, here one of the keys
+    // will be used without considering PropertyType
+    // TODO each table will keep a separate instance list.This can be improve by only keeping one copy of the data
```

Comment:
Thanks for adding this todo.
But for the logic, it is too verbose. Why not just get the _routingTableRefMap.values().interator().next()?

Reply:
Changed based on your suggestion. Thanks.